src/Hosting.CommandLine/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,69 @@
+using McMaster.Extensions.CommandLineUtils;
+using McMaster.Extensions.Hosting.CommandLine.Internal;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System.Threading.Tasks;
+
+namespace McMaster.Extensions.Hosting.CommandLine
+{
+    /// <summary>
+    /// Extension methods for <see cref="IHostBuilder"/> support.
+    /// </summary>
+    /// <seealso href="https://github.com/natemcmaster/CommandLineUtils/issues/134">host support</seealso>
+    public static class HostBuilderExtensions
+    {
+        /// <summary>
+        /// Runs <code>T</code> as a <see cref="CommandLineApplication"/> with the
+        /// supplied <code>args</code>.  This method should be the primary approach 
+        /// taken for command line applications.
+        /// </summary>
+        /// <typeparam name="T">The command line application implementation</typeparam>
+        /// <param name="hostBuilder">This instance</param>
+        /// <param name="args">The command line arguments</param>
+        /// <returns>A task whose result is the exit code of the application</returns>
+        public static async Task<int> RunCliAsync<T>(this IHostBuilder hostBuilder, string[] args) where T : class
+        {
+            using (var host = hostBuilder.UseCliLifetime().UseCli<T>(args).Build()) 
+            {
+                await host.StartAsync();
+                await host.WaitForShutdownAsync();
+                return ((CliLifetime)host.Services.GetService<IHostLifetime>()).ExitCode;
```

Comment:
Casting is bad. Can we find another way to get the exit code?

Reply:
Not sure here.  The `IHostLifetime` seems to be the right place to store this (exit code sounds like something that pertains to the _end_ of a lifecycle).  But `IHostLifetime` has no concept of exit code and is completely out of our control.  

In any case, this extension method ensures that the `IHostLifetime` is set (`UseCliLifetime`) prior to this cast, so it is _safe_ internally.

Other Options:
* Check before casting
  * Not sure what this would accomplish if anything
* Throw an exception on non-zero exit code, catch it here, convert back to int
  * _This seems a little crazy_
* We could store it in the `CliService` and add `ExitCode` as a property to `ICliService
  * Not sure this makes much sense as it is the _result_ of the `RunAsync` so we would have to deal with the value not being set synchronously (not even sure how that would work).  
* Create a _Service_ to carry state, and update it from the CliLifetime (perhaps `CommandLineContext` [if we can figure out why that isn't injecting](https://github.com/natemcmaster/CommandLineUtils/pull/167#discussion_r226874727))
  * This _almost_ makes sense because it is context...

What do you think?

Reply:
I like the last option you mentioned, creating a service that carries state. We can use `CommandLineArgs` for now while we figure out if we can re-use `CommandLineContext`.

Reply:
Renamed `CommandLineArgs` to `CommandLineState` and [used it to propagate the exit code](https://github.com/natemcmaster/CommandLineUtils/pull/167/commits/79f964161903730650a4171d420519a0d4bb4247)